src/cockpit/channels/dbus.py

```diff
@@ -67,6 +159,14 @@ async def get_signature(self, interface_name, method, bus=None, destination=None
         return ''.join(interface['methods'][method]['in'])
 
 
+def notify_update(notify, path, interface_name, props):
+    data = {k: v['v'] for k, v in props.items()}
+    if path in notify:
+        notify[path][interface_name] = data
+    else:
+        notify[path] = {interface_name: data}
```

Comment:
The standard Python idiom for this is:

    notify.setdefault(path, {})[interface_name] = data

Reply:
Done, thanks!